app/scripts/modules/core/src/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx

```diff
@@ -47,6 +78,35 @@ export class ManualJudgmentApproval extends React.Component<
     );
   }
 
+  private isStageAuthorized(): boolean {
+
+      let disableBtn = true;
+      let usrRole;
+      const stageRoles = this.props.stage.context.selectedStageRoles || [];
+      const readArray = this.state.applicationRoles['READ'] || [];
+      const writeArray = this.state.applicationRoles['WRITE'] || [];
+      const executeArray = this.state.applicationRoles['EXECUTE'] || [];
+      const createArray = this.state.applicationRoles['CREATE'] || [];
+      const usrRoles = this.state.userRoles;
+      if (stageRoles.length === 0) {
+          disableBtn = false;
+          return disableBtn;
+      }
+      for (usrRole of usrRoles) {
+          if (stageRoles.indexOf(usrRole) > -1) {
+              if (writeArray.indexOf(usrRole) > -1 ||
+                  executeArray.indexOf(usrRole) > -1 ||
+                  createArray.indexOf(usrRole) > -1) {
+                  disableBtn = false;
+                  break;
+              } else if (readArray.indexOf(usrRole) > -1) {
+                  disableBtn = true;
+              }
```

Comment:
This is a little tough to parse; if I'm understanding correctly, the user _is_  authorized to approve the Manual Judgment if the following is true: The `selectedStageRoles` (which are the roles that were configured against the Manual Judgment stage config) includes at least one of the roles of the authorized user, and that role must also be of one of the following types for the application (WRITE, EXECUTE, CREATE). Am I understanding correctly?

If this is the case, why are we hydrating the available roles users can select when configuring the Manual Judgment stage with (READ, WRITE, EXECUTE) roles if READ is now insufficient to approve a Manual Judgment?

Sorry if these are ignorant questions; I haven't played around much with authz in Spinnaker.

From a code perspective, can be more readable in TS to use [`includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) rather than checking  `indexOf > -1`, and [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) rather than for/of with a `break` to stop after some condition is reached for an element of a collection.

Finally, per my other comment, it's a bit confusing to return `disableBtn` from this method, since it is called `isStageAuthorized`. (I find it easier to use consistently positive naming.)

Reply:
Yes , READ is insufficient to approve a Manual Judgment.

Reply:
Modified the code with includes approach.